core/kazoo_documents/src/kzd_fax.erl

```diff
@@ -226,3 +249,411 @@ retry_after(FaxDoc) ->
 -spec retry_after(doc(), Default) -> integer() | Default.
 retry_after(FaxDoc, Default) ->
     kz_json:get_integer_value(?KEY_RETRY_AFTER, FaxDoc, Default).
+
+%%%=============================================================================
+%%% attachment handling functions
+%%%=============================================================================
+
+%%------------------------------------------------------------------------------
+%% @doc faxes pre-convert attachment documents to tiff/pdf on ingress and save
+%% all these formats to the db as attachments
+%%
+%% If configured, the fax document for outbound faxes will contain a copy of the
+%% post conversion fax tiff file and a pdf representation of this file.
+%%
+%% @end
+%%------------------------------------------------------------------------------
+-spec save_outbound_fax(kz_term:ne_binary(), kz_json:object(), kz_term:api_binary(), kz_term:api_binary()) ->
+                               {'ok', kz_json:object()} |
+                               {'error', any()}.
+save_outbound_fax(Db, Doc, 'undefined', _) ->
+    case fetch_attachment_url(Doc) of
+        {'ok', Content, ContentType} ->
+            case kapps_config:get_is_true(?FAX_CONFIG_CAT, <<"store_url_document">>, true) of
+                'true' -> save_outbound_fax(Db, Doc, Content, ContentType);
+                'false' -> {'ok', Doc}
+            end;
+        Error -> Error
+    end;
+save_outbound_fax(Db, Doc, Original, ContentType) ->
+    Id = kz_doc:id(Doc),
+    Name = <<?ORIGINAL_FILE_PREFIX, ".", (kz_mime:to_extension(ContentType))/binary>>,
+    Att = [{Original, ContentType, Name}],
+    case kapps_config:get_is_true(?FAX_CONFIG_CAT, <<"store_fax_tiff">>, true) of
+        'true' ->
+            case convert_to_fax(ContentType, Original, Id) of
+                {'ok', Tiff, Props} ->
+                    NewDoc = update_fax_props(Doc, Props),
+                    Att1 = Att ++ [{Tiff, ContentType, ?FAX_FILENAME}],
+                    save_fax_attachments(Db, NewDoc, Att1 ++ [maybe_convert_to_pdf(Tiff, kz_doc:id(NewDoc))]);
```

Comment:
You can write `[Att, {Tiff, ContentType, ?FAX_FILENAME} | maybe_convert_to_pdf(Tiff, kz_doc:id(NewDoc))]` and return list in `maybe_convert_to_pdf` (if it is not converting then just return an empty list).